app/controllers/user.controller.js

```diff
@@ -106,21 +106,16 @@ exports.findByEmail = (req, res) => {
 // Update a User by the id in the request
 exports.update = async (req, res) => {
   const id = req.params.id;
-  if (!validateRole(req.body.role)) {
-    res.status(400).send({ message: invalidRole })
-    return;
+  let userForEmail = undefined
+  if (req.body.email) {
+    userForEmail = await getUserForEmail(req.body.email)
```

Comment:
```suggestion
  if (req.body.email) {
    let userForEmail = await getUserForEmail(req.body.email)
```
No need for larger scope.